GetStream/stream-chat-swift
# Fix redundant API call in ChannelListController

### 🎯 Goal

Once, we had a trumping channel logic, which was a workaround for outdated data. It was problematic so we disabled it (by commenting out the code) but we didn't remove the whole fix, so the SDK is making 1 extra API call because of this.

### 🛠 Implementation

Tthe connection observer and the extra API call is removed from `ChannelListController`

Additionally, I found some broken tests when I ran the tests locally, so I fixed them in a separate commit. I don't know how CI was green on #1642 , perhaps CI can't import the other SDKs but local builds can?

### 🧪 Testing

- Launch the app with a proxy
- Observe the API calls.
There should be 2 `queryChannels` call, 1 is the expected one, the other will be resolved by offline storage PRs.

### ☑️ Checklist
